Automattic/pocket-casts-ios
# Tracks: Adds table swipe action analytics

| 📘 Project: #154 |
|:---:|

Adds analytics when a user uses one of the actions presented in the table swipe actions:
- `episode_swipe_action_performed`: When a user taps on one of the actions presented after swiping a Podcast episode row	"
    - `source`: String - The view that is hosting the episode rows. Possible values: `[downloads, listening_history, filters, podcast_list, starred, files]`
    - `action`: String - The name of the action being performed. Possible values: `[up_next_remove, up_next_add_top, up_next_add_bottom, archive, unarchive, delete, up_next_move_up, up_next_move_down]`

## To test

1. Launch the app
2. Go to the Podcasts tab
3. Tap on any podcast to view the episodes
4. Swipe right on an episode to show the left actions
5. Tap on the add to the top of the next up next item: 
6. ✅  `🔵 Tracked: episode_swipe_action_performed ["source": "podcast_list", "action": "up_next_add_top"]`
7. Swipe right again, tap on the remove action
8. ✅ `🔵 Tracked: episode_swipe_action_performed ["action": "up_next_remove", "source": "podcast_list"]`
9. Swipe right again, tap on the add to the bottom action
10. ✅ `🔵 Tracked: episode_swipe_action_performed ["action": "up_next_add_bottom", "source": "podcast_list"]`
11. Swipe left to show the right actions
12. Tap the archive action
13. ✅ `🔵 Tracked: episode_swipe_action_performed ["source": "podcast_list", "action": "archive"]`
14. Tap show archived
15. Swipe left on the archived items
16. Tap on the unarchived action
17. ✅ `🔵 Tracked: episode_swipe_action_performed ["source": "podcast_list", "action": "unarchive"]`
18. Repeat on the Downloads list and verify the source is `downloads`
19. Repeat on the Uploaded Files list and verify the source is `uploaded_files`
20. Repeat on the Listening History and verify the source is `listening_history`
21. Add some episodes to the up next queue
22. Tap on the Up Next icon in the miniplayer to view the queue
23. Swipe to the left and tap on the first item (move up)
24. ✅ `🔵 Tracked: episode_swipe_action_performed ["action": "up_next_move_up", "source": "up_next"]`
25. Swipe left again and tap on the second item (move down)
26. ✅  `🔵 Tracked: episode_swipe_action_performed ["action": "up_next_move_down", "source": "up_next"]`
27. Swipe right and tap the delete action
28. ✅  `🔵 Tracked: episode_swipe_action_performed ["source": "up_next", "action": "delete"]`


## Checklist
